> It _is_ related.

Okay, then why does the CLI work perfectly fine outside of the VM? Clearly the test itself is broken and it's unrelated to bw-cli. Did you try running the CLI yourself?



> That failure is a different error

https://logs.ofborg.org/?key=nixos/nixpkgs.379124&attempt_id=278887ba-25ad-48b8-a8a5-8054bfe712cd
```
server # [   33.288443] vaultwarden[1045]: [2025-02-03 16:15:34.857][request][INFO] POST /identity/connect/token
server # [   33.299396] vaultwarden[1045]: [2025-02-03 16:15:34.868][vaultwarden::api::identity][ERROR] Username or password is incorrect. Try again. IP: 2001:db8:1::1. Username: meow@example.com.
server # [   33.307256] vaultwarden[1045]: [2025-02-03 16:15:34.876][response][INFO] (login) POST /identity/connect/token => 400 Bad Request
client # Username or password is incorrect. Try again
(finished: must fail: bw --nointeraction --raw login meow@example.com also_super_secret_ZJWpBKZi668QGt, in 5.66 seconds)
(finished: subtest: can't login to nonexistent account, in 5.66 seconds)
subtest: use the web interface to sign up, log in, and save a password
```

https://logs.ofborg.org/?key=nixos/nixpkgs.378937&attempt_id=28e0439b-52ef-47c9-ae8f-75a7c86a757f

```
server # [   28.510489] vaultwarden[855]: [2025-02-09 21:35:21.967][request][INFO] POST /identity/accounts/prelogin
server # [   28.526156] vaultwarden[855]: [2025-02-09 21:35:21.982][response][INFO] (prelogin) POST /identity/accounts/prelogin => 200 OK
server # [   29.063931] vaultwarden[855]: [2025-02-09 21:35:22.521][request][INFO] POST /identity/connect/token
server # [   29.074763] vaultwarden[855]: [2025-02-09 21:35:22.532][vaultwarden::api::identity][ERROR] Username or password is incorrect. Try again. IP: 2001:db8:1::1. Username: meow@example.com.
server # [   29.084265] vaultwarden[855]: [2025-02-09 21:35:22.542][response][INFO] (login) POST /identity/connect/token => 400 Bad Request
client # Username or password is incorrect. Try again
(finished: must fail: bw --nointeraction --raw login meow@example.com also_super_secret_ZJWpBKZi668QGt, in 6.16 seconds)
(finished: subtest: can't login to nonexistent account, in 6.16 seconds)
subtest: use the web interface to sign up, log in, and save a password
```

How exactly are these errors different?
bw-cli isn't used to create the account, so how is this bw-cli's fault?
The selenium test is clearly broken in some fashion.

--- a/pkgs/by-name/bi/bitwarden-cli/package.nix
+++ b/pkgs/by-name/bi/bitwarden-cli/package.nix
@@ -13,13 +13,13 @@
 
 buildNpmPackage rec {
   pname = "bitwarden-cli";
-  version = "2024.12.0";
+  version = "2025.1.2";
 
   src = fetchFromGitHub {
     owner = "bitwarden";
     repo = "clients";
     rev = "cli-v${version}";
-    hash = "sha256-3aN2t8/qhN0sjACvtip45efHQJl8nEMNre0+oBL1/go=";
+    hash = "sha256-Ibf25+aaEKFUCp5uiqmHySfdZq2JPAu2nBzfiS4Sc/k=";
   };
 
   postPatch = ''
@@ -29,7 +29,7 @@ buildNpmPackage rec {
 
   nodejs = nodejs_20;
 
-  npmDepsHash = "sha256-EtIcqbubAYN9I9wbw17oHiVshd3GtQayFtdgqWP7Pgg=";
+  npmDepsHash = "sha256-+LpF5zxC4TG5tF+RNgimLyEmGYyUfFDXHqs2RH9oQLY=";
 
   nativeBuildInputs = lib.optionals stdenv.hostPlatform.isDarwin [
     cctools
@@ -49,6 +49,14 @@ buildNpmPackage rec {
   npmWorkspace = "apps/cli";
 
   npmFlags = [ "--legacy-peer-deps" ];
+
+  # FIXME temporarily disable the symlink check as there are symlink after the build in the `node_modules/@bitwarden` directory linking to `../../`.
+  dontCheckForBrokenSymlinks = true;
+
+  npmRebuildFlags = [
+    # FIXME one of the esbuild versions fails to download @esbuild/linux-x64
+    "--ignore-scripts"
+  ];
 
   postConfigure = ''
     # we want to build everything from source

